multihash.go

```diff
@@ -222,20 +222,34 @@ func Cast(buf []byte) (Multihash, error) {
 
 // Decode parses multihash bytes into a DecodedMultihash.
 func Decode(buf []byte) (*DecodedMultihash, error) {
-	rlen, code, hdig, err := readMultihashFromBuf(buf)
+	// outline decode allowing the &dm expression to be inlined into the caller.
+	// This moves the heap allocation into the caller and if the caller do not
```

Comment:
*doesn‘t

Reply:
https://github.com/multiformats/go-multihash/pull/174/files#diff-12ca94b89a37cc5c4cf450ef23cee795d96bbf54a89650c1ad58caede6b33ee9R346
```
new alloc/op
        0.0B
```
There is no "`IsStackAllocated`" function that is reliable.

Reply:
Done